In [10]:
import numpy as np
import scipy.stats as stats

# Домашняя работа по уроку 5. Тестирование гипотез

## Задача 1

Когда используется критерий Стьюдента, а когда Z –критерий?

### Решение

Z-критерий используется для проверки гипотез относительно математического ожидания 

нормально распределенной случайной величины с известной дисперсией.

В предположении верности нулевой гипотезы z-статистика имеет 

стандартное нормальное распределение (0, 1).

Z = (X<sub>cp</sub> - mu) / (std/sqrt(n))

Критерий Стьюдента (t-критерий) используется, когда дисперсия неизвестна.

В предположении верности нулевой гипотезы t-статистика имеет распределение Стьюдента.

t = (X<sub>cp</sub> - mu) / (std<sub>x</sub>/sqrt(n))

---------------------

## Задача 2

Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные

автоматическим станком, имеют средний диаметр 17 мм.

Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из

n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

### Решение

1) Формулируем гипотезы:

H<sub>0</sub> : mu = 17 - предполагаем, что средний диаметр не менялся и составляет 17мм.

H<sub>1</sub> : mu > 17 - предполагаем, что средний диаметр увеличился и теперь он больше 17мм.

2) α=0,05 - статистический уровень значимости обозначен условиями задачи.

3) Нам известна дисперсия, значит, будем использовать для проверки гипотезы Z-критерий:

Найдем наблюдаемое значение Z-критерия:

In [5]:
mu = 17
x = 17.5
a = 0.05
n = 100
std = 4 ** 0.5
Z = (x - mu) / (std/(n ** 0.5))
Z

2.5

Из z-таблицы нормального распределения найдем 

критическое значение Z-критерия, исходя из значения α=0,05

Z<sub>кp</sub> = 1.65

In [ ]:
#            ////////////////////////
# ----------|----------|-------------
#          Zкр         Zн
#          1.65        2.5

4) Таким образом, видим, что наше наблюдаемое значение находится в области критического значения.

Следовательно, мы отвергаем гипотезу H<sub>0</sub>, и принимаем гипотезу H<sub>1</sub> о том, 

что средний диаметр шариков больше 17мм.

------------

## Задача 3

Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г.

Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190.

Известно, что их веса распределены нормально.

Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна

99%? (Провести двусторонний тест.)

### Решение

1) Формулируем гипотезы:

H<sub>0</sub> : mu = 200 - предполагаем, что средний вес пачки составляет 200г.

H<sub>1</sub> : mu != 200 - предполагаем, что средний вес пачки не равкен 200г.

2) α=0,01 (100% - 99%) - статистический уровень значимости обозначен условиями задачи.

3) Нам не известна дисперсия, значит, будем использовать для проверки гипотезы t-критерий:

Найдем наблюдаемое значение t-критерия:

In [7]:
mu = 200
n = 10
arr = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
alpha = 0.01

# Среднее значение по выборке
mean = np.mean(arr)
mean

198.5

In [8]:
# Среднее квадратичное отклонение по выборке
std = np.std(arr, ddof=1)
std

4.453463071962462

In [9]:
# Наблюдаемое значение t-критерия
t = (mean - mu) / (std / np.sqrt(n))
t

-1.0651074037450896

Найдем табличное значение критерия Стьюдента для выборки из 10 пачек и α=0,01

t<sub>кр</sub> = 3,250

In [ ]:
# \\\\\\\\\\                          /////////////
# ----------|----------|-------------|-------------
#          tкр        tн            tкр
#        -3.25      -1.06           3.25

3) Таким образом, видим, что наблюдаемое значение t-критерия расположено 

за пределами области критического значения t-критерия.

Поэтому, мы считаем верной гипотезу H<sub>0</sub> - средний вес пачки печенья составляет 200г.

----------

## Задача 4

Есть ли статистически значимые различия в росте дочерей?

Рост матерей 172, 177, 158, 170, 178, 175, 164, 160, 169, 165

Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160, 163

### Решение

Проводим двухвыборочный тест, т.к. имеем две выборки.

Данные выборки являются зависимыми, т.к. есть родственные связи между группами выборок,

а, значит, рост матерей может влиять генетически на рост дочерей.

Формулируем гипотезы:

H<sub>0</sub> : mu<sub>x</sub> = mu<sub>y</sub> - предполагаем, что различий нет.

H<sub>1</sub> : mu<sub>x</sub> != mu<sub>y</sub> - предполагаем, что различия есть.

α=0,05 - стандартное значение.

In [13]:
# выборки 
x = np.array([172, 177, 158, 170, 178, 175, 164, 160, 169, 165])
y = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160, 163])

In [20]:
# Используем функцию для проверки гипотезы с зависимыми выборками
stats.ttest_rel(x, y)

TtestResult(statistic=0.6648478531431979, pvalue=0.5228168632983574, df=9)

Таким образом, p-value = 0.52 > α = 0,05, значит,

гипотеза H<sub>0</sub> верна и нет статистически значимых различий в росте дочерей.

Иными словами, рост дочерей примерно такой же, как рост матерей, как результат наследственности.